File: EPI/11.5/typescript/11.5_test.ts
import medians from './11.5';

declare function require(name: String): any;
const assert = require('assert');
const inspect = (v: any) => require('util').inspect(v, {depth: null});

const tests: {values: number[], medians: number[]}[] = [
    // {
    //     values: [],
    //     medians: []
    // },
    // {
    //     values: [1],
    //     medians: [1]
    // },
    {
        values: [1, 0, 3, 5, 2, 0, 1],
        medians: [1, 0.5, 1, 2, 2, 1.5, 1]
    },
];

for (const test of tests) {
    const actual = medians(test.values);
    assert.deepStrictEqual(actual, test.medians, `For values ${inspect(test.values)}, ` +
        `expected medians to be ${test.medians}, but were ${actual}`);
}

File: EPI/Heaps/Compute the median of online data/swift/11.5_test.swift
import Darwin

let tests: [(values: [Int], medians: [Float])] = [
    (
        values: [],
        medians: []
    ),
    (
        values: [1],
        medians: [1]
    ),
    (
        values: [1, 0, 3, 5, 2, 0, 1],
        medians: [1, 0.5, 1, 2, 2, 1.5, 1]
    ),
]

for test in tests {
    let actual = medians(test.values)
    guard actual == test.medians else {
        print("For test values \(test.values), expected medians to be \(test.medians), " +
            "but were \(actual)")
        exit(1)
    }
}
